utils/modular_model_converter.py

```diff
@@ -700,7 +755,44 @@ def leave_If(self, original_node, node):
                 self.new_body[node] = {"insert_idx": self.global_scope_index, "node": node}
         return node
 
-    def leave_Module(self, original_node: cst.Assign, node):
+    def visit_Call(self, node: cst.Call):
+        """This is used to create a mapping from functions to class calling them, and from top-level functions to functions called inside them"""
+        # Only map function calls if we're inside a class (i.e., current_class is set)
+        if self.current_class is not None:
+            # Simple function calls such as foo()
+            if isinstance(node.func, cst.Name):
+                self.function_call_class_mapping[node.func.value].add(self.current_class)
+        elif self.current_top_level_function is not None:
+            # Simple function calls such as foo()
+            if isinstance(node.func, cst.Name):
+                self.function_call_dependency_mapping[self.current_top_level_function].add(node.func.value)
+
+    def _add_function_to_body(
+        self,
+        top_level_function: str,
+        body: dict,
+        function_node: cst.FunctionDef,
+        matching_callers: set | None = None,
+        parent: str | None = None,
+    ) -> bool:
+        """Add the given function to the body in the correct place if it not already present."""
```

Comment:
To update this is more like "returns wether or not the function node should be added to the updated body. Will be True if the function is called anywhere, false otherwise